tools/faucet/faucet/faucet.go

```diff
@@ -85,7 +83,7 @@ func (f *Faucet) validateTx(tx *types.Transaction) error {
 	for now := time.Now(); time.Since(now) < _timeout; time.Sleep(time.Second) {
 		receipt, err := f.client.TransactionReceipt(context.Background(), tx.Hash())
 		if err != nil {
-			if errors.Is(err, rpc.ErrNilResponse) {
+			if receipt == nil {
```

Comment:
You still need to check the type of error here is `ethereum.NotFound` I think? Else it'll keep waiting when the node is down for example.

Reply:
good spot. this logic is dodgy here